app/assets/javascripts/controllers/ops/pglogical_replication_form_controller.js

```diff
@@ -39,14 +39,15 @@ ManageIQ.angular.app.controller('pglogicalReplicationFormController', ['$http',
 
   $scope.saveClicked = function() {
     // remove existing subscriptions that have not changed before sending them up for save
-    $scope.pglogicalReplicationModel.subscriptions.forEach(function(subscription, index, object) {
+    var modified_subscriptions = $scope.pglogicalReplicationModel.subscriptions.slice();
+    modified_subscriptions.forEach(function(subscription, index, object) {
```

Comment:
The change makes sense, but if you're rewriting it, maybe we should get rid of the whole "lets foreach an array and remove entries from it while looping" in favor of something like `filter`.

What about this..

```
var modified_subscriptions = $scope.pglogicalReplicationModel.subscriptions.filter(function(subscription, index) {
  // new
  if (typeof subscription.id === 'undefined') {
    return true;
  }

  // removed
  if (subscription.remove === true) {
    return true;
  }

  // changed
  if (subscriptionChanged(subscription, $scope.modelCopy.subscriptions[index])) {
    return true;
  }

  // unchanged
  return false;
});
```

Reply:
(Or just `var modified_subscriptions = $scope.pglogicalReplicationModel.subscriptions.filter(function(subscription, index) { return (! subscription.id) || subscription.remove || subscriptionChanged(subscription, $scope.modelCopy.subscriptions[index]); };` :).)

Reply:
Done. Thanks!